src/compiler/emitter.ts

```diff
@@ -4128,17 +4148,19 @@ module ts {
 
             function getLeadingCommentsToEmit(node: Node) {
                 // Emit the leading comments only if the parent's pos doesn't match because parent should take care of emitting these comments
-                if (node.parent.kind === SyntaxKind.SourceFile || node.pos !== node.parent.pos) {
-                    var leadingComments: CommentRange[];
-                    if (hasDetachedComments(node.pos)) {
-                        // get comments without detached comments
-                        leadingComments = getLeadingCommentsWithoutDetachedComments();
-                    }
-                    else {
-                        // get the leading comments from the node
-                        leadingComments = getLeadingCommentRangesOfNode(node, currentSourceFile);
+                if (node.parent) {
+                    if (node.parent.kind === SyntaxKind.SourceFile || node.pos !== node.parent.pos) {
+                        var leadingComments: CommentRange[];
+                        if (hasDetachedComments(node.pos)) {
+                            // get comments without detached comments

```

Comment:
Useless message. What is a detached comment and where is it documented?

Also, I'm assuming that in `hasDetachedComments`, `detachedCommentsInfo` is only initialized upon having an element. If so, use `lastOrUndefined` there.